### ISSUE TITLE
Opening email from GMAIL sent folder opens corresponding IMAP uid from INBOX (ver 7.9.0)

### ISSUE THREAD

**Issue by dawansv** [salesagility/SuiteCRM#3646]
#### Issue
When viewing the detail of an email from GMAIL sent folder, it opens the email from the inbox with the same IMAP uid.

It seems the IMAP uid provided is not unique across folder. It is unclear to me whether this is a GMAIL issue only. I also found this reference to a unique message ID.
https://developers.google.com/gmail/imap/imap-extensions#access_to_the_gmail_unique_message_id_x-gm-msgid

#### Expected Behavior
Should open email from the sent items

#### Actual Behavior
Email from inbox is shown instead

#### Possible Fix
Should the uid be unique accross all folders? For now it isn't. So it is unclear whether the problem originates here or with GMAIL.

If the folder matters, then a cursory look at the URL shows that is has 'folder=' twice in the query string, one pointing to the imap folder, one pointing to the type. This seems a bit strange.
`index.php?module=Emails&action=DisplayDetailView&folder=[Gmail]/Sent%20Mail&folder=sent&inbound_email_record=7f836cc4-3df4-bc33-7c7a-59306a615e68&uid=24&msgno=24`

Randomly changing the value of folder in the querystring still returns the same email, so it seems the folder information is ignored somewhere in the code and INBOX used as default. So the only thing that seems to matter is inbound_email_record and uid

#### Steps to Reproduce
1. With a Gmail/Google Apps account, define the sent folder in the Mail Settings as [Gmail]/Sent Mail
2. Go to emails, click on INBOX button on the right, open list with arrow, select Sent folder
3. Try opening an email from the sent folder

#### Context
If this is only a GMAIL issue, then as a workaround we should tell users to connect the Inbox (as defined by SuiteCRM) to the [Gmail]/All Mail folder instead of the Gmail inbox folder; then leave the sent folder blank. This way all mail will be available from the main folder. This is what we are doing for now.

#### Your Environment
* SuiteCRM Version used: 7.9
* Browser: Chrome Version Version 58.0.3029.110 (64-bit)
* Environment name and version : MySQL 5.7.18, PHP 5.6
* Operating System and version: Ubuntu 16.04

**Comment by adriangibanelbtactic** [comment]
Your link: https://developers.google.com/gmail/imap/imap-extensions#access_to_the_gmail_unique_message_id_x-gm-msgid shows what I already knew about Gmail.

If you add many tags to a single email then the same email (repeated) is found via IMAP in as many folders as tags added to that email.

So, yes, from Gmail point of view given the same UID can be found on different folders.

I understand your problem if the email you are checking it's an email sent to you from your account. I mean, it's an email that makes sense being at Inbox and at Sent folders at the same time.

**Comment by dawansv** [comment]
@adriangibanelbtactic 

Indeed Imap folders are just tags from gmail's perspective, and so the same email can be in several folders.

The problem here is different though.  The problem is that the uids used by suitecrm are only unique within a given tag/folder, so you will find different emails with the same uid in different folders. The uids for a given tag/folder seem to be given on a first come first served basis. 

So (unless they have been deleted), I can find one email with uid=34  in my inbox (when accessed from suitecrm) and a different email with uid=34 in my sent mail. Since suitecrm currently defaults to the inbox when opening a message, when I open message with uid=34 from the sent items, I actually get a completely different message that has uid=34 in the inbox.

I assume suitecrm doesn't create the uid but instead gets them from the imap response. What I don't know is whether the "uid is unique within folder only" is a general imap implementation issue or only gmail's version of it.  I don't have access to a non gmail imap server, so somebody else would need to test that.

As per the docs, Google also provide unique ids as per the docs above, but these unique ids are not the ones currently retrieved by suitecrm. But if this is a google implementation only, then would need to treat google imap different from regular imap, which is not ideal of course.

So basically next step (I think) is for somebody with a non-gmail imap to test whether the "uid is unique within folder only" is also a problem...
